### PR TITLE
Profile and Delete Page functional

### LINKED ISSUE #11: ADD profile button in home page and home page
On the profile page, the users can change their preferences also give the option to delete their account (If they are no more interested in receiving reminder mail.)

### PR DESCRIPTION
**Pull request by matt-ross16**
Fixes #11 

There is now a Profile Page available for users that log in to adjust their email preferences (will stay on Profile when submitting).
Users can also delete their account if they wish (will be logged out and redirected to the `Home` view afterwards.

### CODE REVIEW

File: contestreminder/templates/user/profile.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+			{% for field in form.visible_fields %}
+		    	{{ field.label_tag }}
+		    	<span class="imp">{{ field }}</span></br></br>
+				{{ field.errors }}
+			{% endfor %}
+
+			<a class="btn2" href="{% url 'home' %}">Home</a> </br></br>
+
+			<button class="btn1" type="submit">Update</button> &emsp;
+			<a class="btn1" href="{% url 'delete' %}">Delete Account</a>
```
Comment (akshitdesai) [inline contestreminder/templates/user/profile.html]: 1) Please, place the home button below the Update and delete the account buttons.
Also, the Home and Delete update button should be after </form> tag.
2) give btn2 class to Delete account.

File: contestreminder/users/urls.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
-from .views import SignUpView
+from django.contrib.auth.decorators import login_required
+from .views import SignUpView, ProfileView, DeleteUserView
 
 urlpatterns = [
     path('signup/', SignUpView.as_view(), name='signup'),
-]
\ No newline at end of file
+    path('profile/', login_required(ProfileView.as_view()), name='profile'),
+    path('delete/', login_required(DeleteUserView.as_view()), name='delete'),
```
Comment (akshitdesai) [inline contestreminder/users/urls.py]: 1) pass login_url='/users/login' as an argument.

If you will try to access profile page then you will see something like this 
"http://127.0.0.1:8000/accounts/login/?next=/users/profile/"

By default, login_url is /accounts/login so we need to change that.

File: contestreminder/users/views.py [hunk 3]
```diff
@@ -3,15 +3,33 @@
...
     template_name = 'registration/signup.html'
 
+class ProfileView(UpdateView):
+    model = CustomUser
+    form_class = CustomUserChangeForm
+    success_url = reverse_lazy('profile')
+    template_name = 'user/profile.html'
+
+    def get_object(self):
+        return self.request.user
```
Comment (akshitdesai) [inline contestreminder/users/views.py]: 1) specify fields as
fields= ['codeChef', 'codeForces', 'hackerEarth', 'hackerRank', 'spoj']

As we don't want user to see password field in update view for now.

**Review by akshitdesai — CHANGES_REQUESTED** [review]
Please do the changes as required.

File: contestreminder/templates/user/profile.html (hunk 1 above)
Comment (matt-ross16) [reply contestreminder/templates/user/profile.html]: "Also, the Home and Delete update button should be after tag."

I don't think I quite understand this part. Could you rephrase it, please?

File: contestreminder/users/urls.py (hunk 2 above)
Comment (matt-ross16) [reply contestreminder/users/urls.py]: Would it be best to have `login_url` passed for both `login_required` function calls (both the `profile/` and `delete/` paths)?

Or should it be just for `delete/`?

File: contestreminder/users/views.py (hunk 3 above)
Comment (matt-ross16) [reply contestreminder/users/views.py]: Done

File: contestreminder/templates/user/profile.html (hunk 1 above)
Comment (akshitdesai) [reply contestreminder/templates/user/profile.html]: after ```</form>``` tag.
Sorry for the typo.

File: contestreminder/users/urls.py (hunk 2 above)
Comment (akshitdesai) [reply contestreminder/users/urls.py]: Yes, we need to pass login_url for both.
